packages/gatsby-theme-docs/README.md

```diff
@@ -1,3 +1,168 @@
+
+    > For test websites the field should not be set.
+
+  - `createNodeSlug` (_optional_): in case you need to have more control over the creation of the page slugs, you can implement this function. This is useful if for example your website has content files in other file system locations and you want to provide a more meaningful URL path.
+
+    ```ts
+    type Options = { node: Node }; // A Gatsby Node
+    type CreateNodeSlugFn = (originalSlug: string, { node }: Options) => string;
+    ```
+
+- `src/content`: this is where you would put your content pages as `*.mdx` files (_see [Writing content pages](#writing-content-pages)_).
+
+- `src/data/navigation.yaml`: this contains the website main navigation links. The structure of the file is a _list of chapters_ as following:
+
+  ```yaml
+  - chapter-title: This is the title
+    beta: false # (optional): will show the beta flag next to the chapter title
+    pagination: false # (optional) will remove this page from the prev/next content pagination
+    pages:
+      - title: The first page
+        path: '/chapter-1/first-page'
+        beta: false # (optional): will show the beta flag next to the page title
+      - title: Another page
+        path: '/chapter-1/another-page'
+      # another page, and so on...
+  - chapter-title: {} # another chapter, and so on...
+  ```
+
+- `src/content/files`: this folder should contain static files that can be referenced within the `*.mdx` content files. For example SVG files, PDF files, etc.
+
+- `src/images`: this folder should contain images that are used within the `*.mdx` content files. Images in this folder are processed and optimized by Gatsby for lazy loading. Supported image formats are `JPEG` and `PNG`.
+
+## Writing content pages
+
+Content pages are located in the `src/content` folder and should be `*.mdx` files.
+
+Pages can be freely organized accordingly to the website structure. Note that the pages URL matches the file system path by default.
+For example, `src/content/getting-started/installation.mdx` results in the URL path `/getting-started/installation`.
+
+Each `*.mdx` content page must contain some metadata in the frontmatter section, which is the leading section of the file within the `---` lines.
+
+```mdx
+---
+title: The title of this page
+beta: false
+---
+
+This is the actual page content.
+```
+
+Supported frontmatter options are:
+
+- `title`: the title of the page. Most of the time is the same value as in the `src/data/navigation.yaml` but it can be longer if needed.
+- `beta`: to indicate if the _beta_ info message should be displayed or not.
+
+### Available elements within markdown files
+
+Besides the standard markdown syntax, the theme provides some extra elements that can be used within the `*.mdx` files.
+
+The elements should be rendered as XML tags, like HTML elements. For example:
+
+```mdx
+<MyComponent>
```

Comment:
I suggest to only mention components that actually exist.  "MyComponent" implies understanding what a "component" is and how it comes about and how you develop it etc.  If you want to keep it generic, rather user "MyElement" because the rest of the text refers to elements.   I would prefer to stick with "Component" instead of "Element" to ease developer<->writer communication

```
<Warning>
```

Reply:
Yeah good point. I wasn't sure to use "component" as maybe "element" might be more understandable for non technical people. Should we just go with the wording "component" then?

Reply:
Yes, I would generally stick with component.  In a separate piece of documentation or training we can explain the relation to HTML and the whole idea behind MDX.   
"Element" could technically spoken also be a custom HTML element and would then have to be lowercased.

Reply:
https://github.com/commercetools/commercetools-docs-kit/pull/90/commits/5ecd9303d365bce0f3044ac30b3f419fedde1182